src/containers/Accounts/AMM/AMMAccounts/index.tsx

```diff
@@ -0,0 +1,140 @@
+
+function findAMMCreate(txs: [any]) {
+  const ammCreate = txs.filter((tx) => tx.tx.TransactionType === 'AMMCreate')
+
+  if (ammCreate.length < 1) throw new Error('Could not find AMM Create')
+
+  return ammCreate[0].tx
+}
+
+export const AMMAccounts = () => {
+  const { id: accountId, tab = 'transactions' } = useParams<{
+    id: string
+    tab: string
+  }>()
+  const { path = '/' } = useRouteMatch()
+  const { t } = useTranslation()
+  const mainPath = `${path.split('/:')[0]}/${accountId}`
+  const rippledSocket = useContext(SocketContext)
+  const [data, setData] = useState<AmmDataType>()
+  const [error, setError] = useState<any>()
+  const language = useLanguage()
+
+  useEffect(() => {
+    let asset1: { currency: string; issuer?: string }
+    let asset2: { currency: string; issuer?: string }
+    let ammData: any
+
+    /*
+    Get the first account transaction which in this case should be AMMCreate. From this we get
+    the two assets in the asset pool.
+    */
+    getAccountTransactions(rippledSocket, accountId, 1, undefined, true)
+      .then((tData) => {
+        const tx = findAMMCreate(tData.transactions)
+        asset1 = formatAsset(tx.Amount)
+        asset2 = formatAsset(tx.Amount2)
+
+        // if one of the assets is XRP, make sure it's the second one
+        if (asset1.currency === 'XRP') {
+          const temp = asset2
+          asset2 = asset1
+          asset1 = temp
+        }
+
+        return getAMMInfo(rippledSocket, asset1, asset2)
+      })
+
+      /*
+      Use the assets to get the AMM Info.
+      */
+      .then((ammDataWrapper) => {
+        ammData = ammDataWrapper.amm
+        const balance = formatBalance(ammData.Amount)
+        const balance2 = formatBalance(ammData.Amount2)
+
+        const ammInfo: AmmDataType = {
+          balance,
+          balance2,
+          tradingFee: ammData.TradingFee,
+          lpBalance: ammData.LPToken.value,
+          accountId,
+          language,
+        }
+
+        setData(ammInfo)
+      })
+      .catch((e) => {
+        analytics(ANALYTIC_TYPES.exception, {
+          exDescription: `Error setting up amm account --- ${JSON.stringify(
+            e,
+          )}`,
+        })
+
+        if (e.message !== 'onthedex failure') setError(e)
+      })
+  }, [accountId, rippledSocket])
+
+  function renderError() {
+    const message = getErrorMessage(error)
+    return (
+      <div className="accounts-page">
+        <NoMatch title={message.title} hints={message.hints} />
+      </div>
+    )
+  }
+
+  useEffect(() => {
+    analytics(ANALYTIC_TYPES.pageview, { title: 'Accounts', path: '/accounts' })
+
+    return () => {
+      window.scrollTo(0, 0)
+    }
+  })
+
+  document.title = `${t('xrpl_explorer')} | ${accountId.substr(0, 12)}...`
+
+  const tabs = ['transactions']
+
+  return error ? (
+    renderError()
+  ) : (
+    <div className="accounts-page section">
+      {data && (
+        <>
+          <AMMAccountHeader {...data!} />
```

Comment:
Why is this using a spread instead of `data={data}`? Our linter doesn't like spreads

Reply:
fixed